### PR TITLE
feat(ui+ffi): Add various features to `RoomList`

### PR DESCRIPTION
**Pull request by Hywan**
Address https://github.com/matrix-org/matrix-rust-sdk/issues/1911.

This PR must be reviewed commit by commit.

### CODE REVIEW

File: bindings/matrix-sdk-ffi/src/room_list.rs [hunk 1]
```diff
@@ -198,6 +198,10 @@ impl RoomListItem {
         RUNTIME.block_on(async { self.inner.name().await })
     }
 
+    fn full_room(&self) -> Arc<Room> {
```
Comment (bnjbvr) [inline bindings/matrix-sdk-ffi/src/room_list.rs:201]: What do you mean by `full`?

File: crates/matrix-sdk-ui/src/room_list/state.rs [hunk 2]
```diff
@@ -89,7 +90,8 @@ impl Action for AddVisibleRoomsList {
             .add_list(
                 SlidingSyncList::builder(VISIBLE_ROOMS_LIST_NAME)
                     .sync_mode(SlidingSyncMode::new_selective())
-                    .timeline_limit(20),
+                    .timeline_limit(20)
+                    .required_state(vec![(StateEventType::RoomEncryption, "".to_owned())]),
```
Comment (bnjbvr) [inline crates/matrix-sdk-ui/src/room_list/state.rs]: Should we also enable `m.room.member` events here, to help address https://github.com/matrix-org/matrix-rust-sdk/issues/2004?

**Review by bnjbvr — COMMENTED** [review]
Looks like it's on the right path! A few questions below.

File: crates/matrix-sdk-ui/src/room_list/state.rs (hunk 2 above)
Comment (Hywan) [reply crates/matrix-sdk-ui/src/room_list/state.rs]: Maybe! @poljar, @stefanceriu, @manuroe, any thoughts?

File: bindings/matrix-sdk-ffi/src/room_list.rs (hunk 1 above)
Comment (Hywan) [reply bindings/matrix-sdk-ffi/src/room_list.rs:201]: It's a legacy term that means: gimme the matrix_sdk_ffi:Room instance. Happy to change to something more meaningful.

File: crates/matrix-sdk-ui/src/room_list/state.rs (hunk 2 above)
Comment (stefanceriu) [reply crates/matrix-sdk-ui/src/room_list/state.rs]: Isn't damir talking only about the rooms you're subscribing to and not the full lists? 🤔

**Review by stefanceriu — APPROVED** [review]
This looks good and works well 👍

File: crates/matrix-sdk-ui/src/room_list/state.rs (hunk 2 above)
Comment (poljar) [reply crates/matrix-sdk-ui/src/room_list/state.rs]: Yes, you're only sending events in a room that you're subscribed to. So only there we'll need the member events.

File: bindings/matrix-sdk-ffi/src/room_list.rs (hunk 1 above)
Comment (bnjbvr) [reply bindings/matrix-sdk-ffi/src/room_list.rs:201]: Ah ok, as a newcomer this feels a bit strange, maybe `room` would be sufficient? (no strong opinion here, so you choose)

**Review by bnjbvr — APPROVED** [review]
Cool, thanks for the answers, lgtm!

**Pull request merged**